src/clp_ffi_js/ir/StreamReader.cpp

```diff
@@ -165,52 +199,92 @@ auto StreamReader::decode_range(size_t begin_idx, size_t end_idx) const -> Decod
         }
         message.append(parsed.value());
 
-        constexpr size_t cLogLevelPositionInMessages{1};
-        size_t log_level{cLogLevelNone};
-        // NOLINTNEXTLINE(readability-qualified-auto)
-        auto const log_level_name_it{std::find_if(
-                cLogLevelNames.begin() + cValidLogLevelsBeginIdx,
-                cLogLevelNames.end(),
-                [&](std::string_view level) {
-                    return message.substr(cLogLevelPositionInMessages).starts_with(level);
-                }
-        )};
-        if (log_level_name_it != cLogLevelNames.end()) {
-            log_level = std::distance(cLogLevelNames.begin(), log_level_name_it);
-        }
-
         m_ts_pattern.insert_formatted_timestamp(log_event.get_timestamp(), message);
 
         EM_ASM(
                 { Emval.toValue($0).push([UTF8ToString($1), $2, $3, $4]); },
                 results.as_handle(),
                 message.c_str(),
                 log_event.get_timestamp(),
-                log_level,
-                log_num
+                log_event.get_log_level(),
+                log_event_idx +1
         );
-        ++log_num;
+
     }
 
     return DecodedResultsTsType(results);
 }
 
+void StreamReader::filter_log_events(const emscripten::val& logLevelFilter) {
+    m_filtered_log_event_map.clear();
+
+    // Check if the filter is empty
+    if (logLevelFilter.isNull()) {
+        m_is_filtered = false;
+        return;
+    }
+
+    // Convert JavaScript array to C++ vector
```

Comment:
Have we tried [vecFromJSArray](https://emscripten.org/docs/api_reference/val.h.html#_CPPv4N10emscripten10emscripten3val14vecFromJSArrayERK3val)?